### PR TITLE
Strassen's Matrix Multiplication

### PR DESCRIPTION
**Pull request by AshishYUO**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: divide_and_conquer/strassen_matrix_multiplication.cpp [hunk 1]
```diff
@@ -0,0 +1,457 @@
+#include <cassert>
+#include <chrono>
+#include <iostream>
+#include <tuple>
+#include <vector>
```
Comment (Panquesito7) [inline divide_and_conquer/strassen_matrix_multiplication.cpp:5]: Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```

File: divide_and_conquer/strassen_matrix_multiplication.cpp [hunk 2]
```diff
@@ -0,0 +1,457 @@
+#include <cassert>
```
Comment (Panquesito7) [inline divide_and_conquer/strassen_matrix_multiplication.cpp:1]: Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..

File: divide_and_conquer/strassen_matrix_multiplication.cpp [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+};
+
+}  // namespace strassens_multiplication
+
+}  // namespace divide_and_conquer
+
+/**
+ * @brief Method for testing current implementation
+ */
+void test() {
```
Comment (Panquesito7) [inline divide_and_conquer/strassen_matrix_multiplication.cpp:414]:
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
static void test() {
```

File: divide_and_conquer/strassen_matrix_multiplication.cpp [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+
+    // std::cout << Mat3 << conf << std::endl;
+    assert(Mat3 == conf);
+}
+
+/**
+ * @brief main function
+ */
+int main() {
+    test();
```
Comment (Panquesito7) [inline divide_and_conquer/strassen_matrix_multiplication.cpp:455]:
```suggestion
 * @brief Main function
 * @returns 0 on exit
 */
int main() {
    test();  // run self-test implementations
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 🚀 Will check the code later on.

**Comment by AshishYUO** [comment]
@Panquesito7 The changes are done. Let me know if it needs more corrections.

**Review by Panquesito7 — APPROVED** [review]
Very good work. Thanks! 🚀
The only thing I saw is that it takes quite a long time to compile (used the Gitpod ready-to-code service). Is this normal?

image

If so, you might want to add a note for that in the code or in a print statement.

**Comment by AshishYUO** [comment]
I usually test the code with optimize flags. I never thought this would actually run that much slow without the flags.
I have reduced the size of the matrix at the moment, it'll now take at most 4-5 seconds.

**Review by tjgurwara99 — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]
Much better on time! Thanks. 🚀

**Pull request merged**